### PR TITLE
 RE-1487 Refactor release pull request validation

### PR DESCRIPTION
**Pull request by git-harry**
Issue: [RE-1487](https://rpc-openstack.atlassian.net/browse/RE-1487)

### CODE REVIEW

File: pipeline_steps/common.groovy [hunk 1]
```diff
@@ -1438,3 +1438,104 @@ Boolean isSkippable(String skip_pattern, String credentials) {
...
+    throw new Exception("The pull request does not correspond to any of ${types}.")
+  }
+
+  println "Pull request is a ${type}."
+  println "=== component CLI standard out ==="
+  println component_text
+  if (type == "release"){
+    testRelease(component_text)
+  }else{
+    throw new Exception("Unknown pull request type ${type}")
```
Comment (dealako) [inline pipeline_steps/common.groovy]: `types` array seems to have two valid values, but we only test for "release".  Perhaps this exception should say "Pull request type ${type} currently not supported" as the previous loop and null check identifies types that are not defined in the array.

Comment (git-harry) [reply pipeline_steps/common.groovy]: Reworded.

Good spot, I missed this when I rewrote my first version.

**Comment by mattt416** [comment]
recheck_issue_link
recheck_xenial_lint_test

File: pipeline_steps/common.groovy [hunk 2]
```diff
@@ -1438,3 +1438,104 @@ Boolean isSkippable(String skip_pattern, String credentials) {
...
+            ]
+          ]
+        )
+      }
+    }
+  }
+
+  parallel parallelBuilds
+
+  build(
```
[the remark this answers is not in the document]
Comment (git-harry) [reply pipeline_steps/common.groovy]: @mattt416 I've split things up a bit to, hopefully, make it clearer. I'm testing it now but take a look and let me know what you think.

**Pull request merged**